ru/lessons/advanced/gen-stage.md

```diff
@@ -0,0 +1,261 @@
+    for event <- events do
+      IO.inspect {self(), event, state}
+    end
+
+    # Так как мы потребители, мы не создаем события
+    {:noreply, [], state}
+  end
+end
+```
+
+Как мы рассмотрели в предыдущем разделе, наш потребитель не создает события, так что второе значение в нашем кортеже будут отброшено.
+
+## Собираем все вместе
+
+Теперь, когда мы создали производителя, производитель-потребителя и потребителя, мы готовы соединить это все вместе.
+
+Давайте откроем файл `lib/genstage_example.ex` и добавим наши новые процессы в дерево надзора:
+
+```elixir
+def start(_type, _args) do
+  import Supervisor.Spec, warn: false
+
+  children = [
+    worker(GenstageExample.Producer, [0]),
+    worker(GenstageExample.ProducerConsumer, []),
+    worker(GenstageExample.Consumer, []),
+  ]
+
+  opts = [strategy: :one_for_one, name: GenstageExample.Supervisor]
+  Supervisor.start_link(children, opts)
+end
+```
+
+Если все правильно, мы можем запустить наш проект и мы должны увидеть что все работает:
+
+```shell
+$ mix run --no-halt
+{#PID<0.109.0>, 2, :state_doesnt_matter}
+{#PID<0.109.0>, 4, :state_doesnt_matter}
+{#PID<0.109.0>, 6, :state_doesnt_matter}
+...
+{#PID<0.109.0>, 229062, :state_doesnt_matter}
+{#PID<0.109.0>, 229064, :state_doesnt_matter}
+{#PID<0.109.0>, 229066, :state_doesnt_matter}
+```
+
+Мы сделали это!  Как мы и ожидали, наше приложение пропускает только четные числа и делает это _быстро_.
+
+На данный момент у нас есть рабочий конвейер.  Производитель генерирует числа, производитель-потребитель отбрасывает нечетные числа и потребитель отображает все это и продолжает процесс. Мы уже упоминали во вступлении, что возможно реализовать более одного производителя или потребителя. Давайте взглянем на это.
+
+Если мы рассмотрим вывод `IO.inspect/1` из нашего приложения, мы увидим, что каждое событие обрабатывается одним PID. Давайте сделаем некоторые корректировки для нескольких работников путем изменения `lib/genstage_example.ex`:
+
+```elixir
+children = [
+  worker(GenstageExample.Producer, [0]),
+  worker(GenstageExample.ProducerConsumer, []),
+  worker(GenstageExample.Consumer, [], id: 1),
+  worker(GenstageExample.Consumer, [], id: 2),
+]
+```
+
+Теперь, когда мы настроили два потребителя, давайте посмотрим что мы получим, если мы запустим наше приложение теперь:
+
+```shell
+$ mix run --no-halt
+{#PID<0.120.0>, 2, :state_doesnt_matter}
+{#PID<0.121.0>, 4, :state_doesnt_matter}
+{#PID<0.120.0>, 6, :state_doesnt_matter}
+{#PID<0.120.0>, 8, :state_doesnt_matter}
+...
+{#PID<0.120.0>, 86478, :state_doesnt_matter}
+{#PID<0.121.0>, 87338, :state_doesnt_matter}
+{#PID<0.120.0>, 86480, :state_doesnt_matter}
+{#PID<0.120.0>, 86482, :state_doesnt_matter}
+```
+
+Как вы можете видеть, теперь мы получили несколько PID, просто добавив строку кода и назначив нашим потребителям ID.
+
+## Случаи использования
+
+Сейчас мы рассмотрели GenStage и построили наш первый пример приложения, какие же есть случаи _реального_ использования GenStage?
```

Comment:
Probably this sentenced would sound better if we split it in two parts like this: 
>Сейчас мы рассмотрели GenStage и построили наш первый пример приложения. А в каких случаях можно  _реально_ использовать GenStage?